Update Location Chapter for matrix (#321)
KhronosGroup/Vulkan-Guide: chapters/location_component_interface.adoc

== Matrix

> A matrix is viewed as an array of vector, this means something like

Reviewer: ```suggestion
A matrix is stored as an array of vector, this means something like
```

"Viewed as" implies something else.

[source,glsl]
----
layout(location = 0) in mat3x2 a;
----

> from a `Location`/`Component` point-of-view looks like

Reviewer: ```suggestion
consumes locations and components identically to
```

layout(location = 0) in vec2 a[3];
----

> Just like vector, depending on the size, it may not consume the whole `Location`. This means the following is allowed.

Reviewer: ```suggestion
which in turn is stored identically to

[source,glsl]
----
// Each vector consumes only the first 2 components of each location
layout(location = 0) vec2 a0;
layout(location = 1) vec2 a1;
layout(location = 2) vec2 a2;
----

This means the following is allowed:
```

Also the "Arrays consume whole location" comment above is wrong - where did that come from?

[source,glsl]
----
layout(location = 0) in mat3x2 a;
layout(location = 1, component = 2) in float b;
layout(location = 2, component = 2) in float c;
----

The `b` and `c` float are still valid here.

image::{images}location_example_matrix.svg[location_example_matrix]

[NOTE]
====

> You don't need to worry about things going "in front" of a matrix because the `Component` decoration
> link:https://godbolt.org/z/f81PvP538[can not be applied on a matrix type].

Reviewer: ```suggestion
Unlike arrays or independent vectors, link:https://godbolt.org/z/f81PvP538[matrices cannot be decorated with `Component`], so will always be packed from component 0. 
```

Maybe? "In front" seems overly vague to me.
